File: GhostTourJPA/src/main/java/com/skilldistillery/ghosttour/entities/Review.java
package com.skilldistillery.ghosttour.entities;

import java.time.LocalDateTime;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity
public class Review {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;

	private String title;
	
	private String name;
	
	private String email;
	
	private String description;
	
	@Column(name="price_per_person")
	private Double pricePerPerson;
	
	private String image;
	
	@Column(name="date_created")
	private LocalDateTime dateCreated;

	public Review() {
		super();
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getTitle() {
		return title;
	}

	public void setTitle(String title) {
		this.title = title;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public Double getPricePerPerson() {
		return pricePerPerson;
	}

	public void setPricePerPerson(Double pricePerPerson) {
		this.pricePerPerson = pricePerPerson;
	}

	public String getImage() {
		return image;
	}

	public void setImage(String image) {
		this.image = image;
	}

	public LocalDateTime getDateCreated() {
		return dateCreated;
	}

	public void setDateCreated(LocalDateTime dateCreated) {
		this.dateCreated = dateCreated;
	}

	@Override
	public String toString() {
		return "Review [id=" + id + ", title=" + title + ", name=" + name + ", email=" + email + ", description="
				+ description + ", pricePerPerson=" + pricePerPerson + ", image=" + image + ", dateCreated="
				+ dateCreated + "]";
	}
	
	

}

File: ngGhostTour/src/app/models/review.ts
import { Tour } from "./tour";

export class Review {

  id: number;
  title: string;
  name: string;
  email: string;
  description: string;
  pricePerPerson: number;
  image: string;
  dateCreated: string;
  tourId: number;
  comments: Comment[];



  constructor(
    id: number = 0,
    title: string = "",
    name: string = "",
    email: string = "",
    decription: string = "",
    pricePerPerson: number = 0,
    image: string = "",
    dateCreated: string = "",
    tourId: number = 0,
    comments: Comment[] = []
  ) {
    this.id = id;
    this.title = title;
    this.name = name;
    this.email = email;
    this.description = decription;
    this.pricePerPerson = pricePerPerson;
    this.image = image;
    this.dateCreated = dateCreated;
    this.tourId = tourId;
    this.comments = comments;
  }
}







